core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -234,17 +263,22 @@ private boolean insertAsSelect(
                     final int cursorTimestampIndex = factory.getMetadata().getColumnIndex(timestampName);
                     assert cursorTimestampIndex > -1;
 
-                    try (RecordCursor cursor = factory.getCursor(refreshExecutionContext)) {
-                        final Record record = cursor.getRecord();
-                        long deadline = batchSize;
-                        rowCount = 0;
-                        while (cursor.hasNext()) {
-                            TableWriter.Row row = tableWriter.newRow(record.getTimestamp(cursorTimestampIndex));
-                            copier.copy(record, row);
-                            row.append();
-                            if (++rowCount >= deadline) {
-                                tableWriter.ic();
-                                deadline = rowCount + batchSize;
+                    long deadline = batchSize;
+                    rowCount = 0;
+
+                    sampleByCursor.toTop();
+                    while (sampleByCursor.next()) {
+                        refreshExecutionContext.setRange(sampleByCursor.getTimestampLo(), sampleByCursor.getTimestampHi());
+                        try (RecordCursor cursor = factory.getCursor(refreshExecutionContext)) {
+                            final Record record = cursor.getRecord();
+                            while (cursor.hasNext()) {
+                                TableWriter.Row row = tableWriter.newRow(record.getTimestamp(cursorTimestampIndex));
+                                copier.copy(record, row);
+                                row.append();
+                                if (++rowCount >= deadline) {
+                                    tableWriter.ic();
```

Comment:
"ic" is old terminology, please use "commit" here

Reply:
Changed in 6a8bb2724b3c827602a8e18b7dce334c0fae434e

Would be nice to add a comment around `ic()` methods as nothing tells that they're obsolete and they're used in INSERT AS SELECT.